### DISCUSSION TITLE
Using RGCNConv with heterogeneous graphs (different node types)

### DISCUSSION THREAD

**Discussion by waterhall** [pyg-team/pytorch_geometric#2939] [Q&A]
Hello, I'm looking for a way to use RGCNConv with heterogeneous graphs (different node types and edges). I understand that RGCNConv can handle different edge types by passing in **edge_type** along with the **edge_index** and **x**.

How can I modify RGCNConv to take into account the different node types? Or is there another existing method that can handle different node types as well as edge types?

Thank you.

**Comment by waterhall** [comment]
I could concatenate a one-hot vector indicating the node type to my current node representation, but I was hoping for a way for the RGCN to have nodetype-specific transformations, similar to how it has different weights for each relation type..

**Reply by rusty1s** [reply]
See https://github.com/rusty1s/pytorch_geometric/issues/2940#issuecomment-892407104
